tools/question-container-debugging.sh

```diff
@@ -0,0 +1,83 @@
+#! /bin/bash
+#
+# Assumes you run this script from your course content directory
+# i.e.
+#   $ ls
+#   courseInstances/ infoCourse.json questions/ serverFilesCourse/
+#   $ ~/git/PrairieLearn/tools/question-container-debugging.sh <QID> [<command to run in container>]
+#
+#   where <QID> is the folder name of the question you want to test
+#   and the last argument is left out to auto-run it, or /bin/bash to explore
+#   the grading container
+#
+# Environment variables:
+#   STUDENTMOUNT="-v /tmp/foo:/grade/student"
+#   PLMOUNT="-v /home/mussulma/git/PrairieLearn:/PrairieLearn"
+
+QID=$1
+if [ -z "$QID" ]; then
+    echo "First argument must be QID (name of question directory)"
+    exit 1
+fi
+
+if [ ! -d "./questions/$QID" ]; then
+    echo "Directory questions/$QID not found, quitting"
+    exit 1
+fi
+
+PLMOUNT=""
+if [ -z "$PLPATH" ]; then
+    echo ""
+    echo "Environment variable PLPATH not set."
+    echo "Set this to the absolute path to your PrairieLearn directory to use it inside the container"
+    echo "Things will likely error without this."
+    echo ""
+else
+    PLMOUNT="-v $PLPATH:/PrairieLearn"
```

Comment:
Do we want to cope with spaces in filenames? If so, this should be
```
PLMOUNT='-v "$PLPATH:/PrairieLearn"'
```
or we should use bash arrays:
```
PLMOUNT=(-v "$PLPATH:/PrairieLearn")
```

Reply:
It's hard to fix that case. I'm having trouble finding an easy way because if we trap the file name (so bash treats it as `'-v /home/mussulma/Prairie Learn:/PrairieLearn'` we still have to pass that into the docker command line which needs escaping. I'm prepared to say "don't have spaces in file names"?